templates/workloads/partials/cf/publish.yml

```diff
@@ -0,0 +1,31 @@
+{{- if .Publish}}
+{{- range $topic := .Publish.Topics}}
+{{$topic.Name}}SNSTopic:
+    Type: AWS::SNS::Topic
+    Properties:
+      TopicName: ${AWS::StackName}-{{$topic.Name}}-SNSTopic
+      
+  {{$topic.Name}}SNSTopicPolicy:
+    Type: AWS::SNS::TopicPolicy
+    DependsOn: {{$topic.Name}}SNSTopic
+    Properties:
+      Topics: 
+        - !Ref ${{.Name}}SNSTopic
+      PolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          {{- range $worker := .AllowedWorkers}}
```

Comment:
Will this syntax error if there are no `AllowedWorkers`? Instead should we have:
```yaml
{{ if $topic.AllowedWorkers }}
{{logicalIDSafe $topic.Name}}SNSTopicPolicy:
  ....
{{- end }}
```

Reply:
Adjusted for an if to prevent syntax errors and changed resource names to include logicalIDSafe.